tastypie/resources.py

```diff
@@ -2406,6 +2406,7 @@ def save(self, bundle, skip_errors=False):
 
         if obj_id not in bundle.objects_saved or bundle.obj._state.adding:
             bundle.obj.save()
+            obj_id = self.create_identifier(bundle.obj)
```

Comment:
I don't understand how this one line makes a difference; the `obj_id` is already created on line 2405.

Reply:
When the first create_identifier function is called the object has not been saved and does not have a id. This makes the return value of that function always have None at the end (ex: `core.mycontaineritemmodel.None`) which is the same as every other unsaved object which causes the saving issue. Once the object is saved the create identifier function will return a unique value.

Reply:
Ah, I see now; on the first call, the PK is `None`, so the `obj_id` ends up shadowing all other not-yet-saved objects.